src/sentry/ratelimits/redis.py

```diff
@@ -15,6 +18,27 @@ class RedisRateLimiter(RateLimiter):
     def __init__(self, **options):
         self.cluster, options = get_cluster_from_options("SENTRY_RATELIMITER_OPTIONS", options)
 
+    def _construct_redis_key(
+        self, key: str, project: Project | None = None, window: int | None = None
+    ) -> str:
+        """
+        Construct a rate limit key using the args given. Key will have a format of:
+        "rl:<key_hex>:[project?<project_id>:]<time_bucket>"
+        where the time bucket is calculated by integer dividing the current time by the window
+        """
+
+        if window is None or window == 0:
+            window = self.window
+
+        key_hex = md5_text(key).hexdigest()
+        bucket = int(time() / window)
+        if project:
+            redis_key = f"rl:{key_hex}:{project.id}:{bucket}"
+        else:
+            redis_key = f"rl:{key_hex}:{bucket}"
```

Comment:
nit: Could be a ternary but that's just because i like how fancy they look, no need to change